packages/ui5-cc-spreadsheetimporter/src/controller/dialog/SpreadsheetUploadDialog.ts

```diff
@@ -299,123 +299,170 @@ export default class SpreadsheetUploadDialog extends ManagedObject {
-		// if sampledata is empty add one row of empty data
-		if (!sampleData || sampleData.length === 0) {
-			sampleData = [{}];
-		}
-		const colWidthDefault = 15;
-		const colWidthDate = 20;
-		let col = 0;
-		let rows = 1;
-		if (this.component.getStandalone()) {
-			// loop over this.component.getColumns
-			for (let column of this.component.getColumns()) {
-				worksheet[XLSX.utils.encode_cell({ c: col, r: 0 })] = { v: column, t: "s" };
-				col++;
-			}
-		} else {
-			for (let [key, value] of this.spreadsheetUploadController.typeLabelList.entries()) {
-				let cell = { v: "", t: "s" } as XLSX.CellObject;
-				let label = "";
-				if (fieldMatchType === "label") {
-					label = value.label;
+	async onTempDownload() {
+		// check if custom template is provided, otherwise generate it
+		if (this.component.getSpreadsheetTemplateFile() !== "") {
+			try {
+				const templateFile = this.component.getSpreadsheetTemplateFile();
+				let arrayBuffer;
+				let fileName;
+
+				if (typeof templateFile === "string") {
+					// Check if the string is a HTTP/HTTPS address
+					if (templateFile.startsWith("http://") || templateFile.startsWith("https://")) {
+						const response = await fetch(templateFile);
+						if (!response.ok) {
+							throw new Error("Network response was not ok " + response.statusText);
+						}
+						fileName = templateFile.split("/").pop();
+						arrayBuffer = await response.arrayBuffer();
+					}
+					// Assume the string is a local file path
+					else {
+						const sPath = sap.ui.require.toUrl(templateFile);
+						const response = await fetch(sPath);
+						if (!response.ok) {
+							throw new Error("Network response was not ok " + response.statusText);
+						}
+						fileName = templateFile.split("/").pop();
+						arrayBuffer = await response.arrayBuffer();
+					}
+				} else if (templateFile instanceof ArrayBuffer) {
+					// If the input is already an ArrayBuffer, use it directly
+					arrayBuffer = templateFile;
+				} else {
+					throw new Error("Unsupported type for templateFile");
 				}
-				if (fieldMatchType === "labelTypeBrackets") {
-					label = `${value.label}[${key}]`;
+
+				// spreadsheet file name wll overwrite the template file name
+				if (this.component.getSpreadsheetFileName() !== "Template.xlsx" || fileName === undefined) {
+					fileName = this.component.getSpreadsheetFileName();
 				}
-				worksheet[XLSX.utils.encode_cell({ c: col, r: 0 })] = { v: label, t: "s" };
 
-				for (const [index, data] of sampleData.entries()) {
-					let sampleDataValue;
-					rows = index + 1;
-					if (data[key]) {
-						sampleDataValue = data[key];
+				Util.downloadSpreadsheetFile(arrayBuffer, fileName);
+
+				// You can now use arrayBuffer to do whatever you need
+			} catch (error) {
+				console.error("Error loading file", error);
+			}
+		} else {
+			// create spreadsheet column list
+			let fieldMatchType = this.component.getFieldMatchType();
+			var worksheet = {} as XLSX.WorkSheet;
```

Comment:
Hey @marianfoo, quick question: is `var` here required for something or is it some leftover?

Reply:
I guess leftover, that is code from the beginning